bigchaindb/__init__.py

```diff
@@ -14,13 +28,7 @@
         'workers': None,  # if none, the value will be cpu_count * 2 + 1
         'threads': None,  # if none, the value will be cpu_count * 2 + 1
     },
-    'database': {
-        'backend': os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'rethinkdb'),
-        'host': os.environ.get('BIGCHAINDB_DATABASE_HOST', 'localhost'),
-        'port': int(os.environ.get('BIGCHAINDB_DATABASE_PORT', 28015)),
-        'name': os.environ.get('BIGCHAINDB_DATABASE_NAME', 'bigchain'),
-        'replicaset': os.environ.get('BIGCHAINDB_DATABASE_REPLICASET', 'bigchain-rs'),
-    },
+    'database': _database_rethinkdb,
```

Comment:
I did it in another pull request https://github.com/bigchaindb/bigchaindb/pull/1088/files#diff-993779d6d8d7ec19551175f387ef2ea6R37

Although your solution looks cleaner. I will apply this changes to #1088